src/shared/credentialsManager.ts

```diff
@@ -49,13 +49,19 @@ export class CredentialsManager {
      * @param profileName Profile to set up
      */
     private async createCredentials(profileName: string): Promise<AWS.Credentials> {
-        const credentials: AWS.Credentials = new AWS.SharedIniFileCredentials({
+        let credentials: AWS.Credentials = new AWS.SharedIniFileCredentials({
             profile: profileName,
             tokenCodeFn: async (mfaSerial, callback) =>
                 await CredentialsManager.getMfaTokenFromUser(mfaSerial, profileName, callback)
         })
 
-        await credentials.getPromise()
+        try {
```

Comment:
Can we get some tests for this?

Reply:
Good point. I'll add some tests to [defaultAwsContext.test.ts](https://github.com/aws/aws-toolkit-vscode/blob/master/src/test/shared/defaultAwsContext.test.ts)

Reply:
@kiiadi - I've played around with this but I haven't found a simple solution for testing. The trouble is, this function is new-ing some AWS SDK objects which have side-effects (like reading from AWS config). Here's a few ideas I've tried for mocking:

(1) Test replaces the AWS object with a fake:
```
    it('returns process crendentials when profile has credentials_process', async () => {
        const settingsConfig = new TestSettingsConfiguration()
        await settingsConfig.writeSetting<string>(profileSettingKey, testProfileValue);
        (AWS as any).SharedIniFileCredentials = function() { return {} }

        const testContext = new DefaultAwsContext(settingsConfig, new FakeExtensionContext())
        const creds = await testContext.getCredentials()
        assert.ok(creds)
    })
```
This test assumes that `new AWS.ProcessCredentials()` will fail. I could also replace `AWS.ProcessCredentials` to make the test more precise but really I'm only testing the logic of `CredentialProviderChain`.

(2) Add `devDependency` on [sinon](https://sinonjs.org/) to manage the mocks/fakes.

My questions are:
1. Now that we're going in the `AWS.CredentialProviderChain` direction, do we still want unit tests?
2. If yes, are you okay with me introducing [sinon] as `devDependency` to make the tests more manageable?
Are you guys opposed to introducing a `devDependency` on `sinon`?

Reply:
We plan to try incorporating jest in the future to help with mocks/fakes, but we don't know yet if that will play well with the VS Code Extension testing host. Using sinon at this time sounds reasonable.

Reply:
Cool. I've updated the PR with tests